src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2583,85 +2583,67 @@ class BacktraceIterator : public StackObj {
 };
 
 
-// Print stack trace element to resource allocated buffer
+// Print stack trace element to the specified output stream.
+// Historically, this was used for java_lang_Throwable and wrote everything to a raw buffer.
+// Presumably to avoid interleaving with other threads.
+// The current version retains that spirit by writing to a stringStream first before
+// flushing everything into the outputStream at once.
```

Comment:
```suggestion
// The output is formatted into a stringStream and written to the outputStream in one step.
```
No need to comment on history as such.